assets/src/edit-story/components/library/panes/text/textSets/karma/textSets.karma.js

```diff
@@ -18,23 +18,75 @@
  * Internal dependencies
  */
 import { Fixture } from '../../../../../../karma/fixture';
+import { useStory } from '../../../../../../app/story';
 
-describe('CUJ: Text Sets (Text and Shape Combinations): Inserting Text Sets', () => {
+describe('Text Sets Library Panel', () => {
   let fixture;
+
   beforeEach(async () => {
     fixture = new Fixture();
+    fixture.setFlags({ showTextSets: true });
+
     await fixture.render();
+
+    // Make Text tab active in library panels
+    const textTab = fixture.editor.library.getByRole('tab', {
+      name: /^Text library$/,
+    });
+
+    await fixture.events.mouse.clickOn(textTab, 10, 20);
   });
 
   afterEach(() => {
     fixture.restore();
   });
 
-  // Disable reason: not implemented yet.
-  // eslint-disable-next-line jasmine/no-disabled-tests
-  xit('should display text sets', async () => {});
+  async function getTextElements() {
+    const {
+      state: {
+        currentPage: { elements },
+      },
+    } = await fixture.renderHook(() => useStory());
+    return elements.filter((e) => e.type === 'text');
+  }
+
+  describe('CUJ: Text Sets (Text and Shape Combinations): Inserting Text Sets', () => {
+    // Disable reason: not implemented yet.
+    // eslint-disable-next-line jasmine/no-disabled-tests
+    xit('should display text sets', async () => {});
+
+    it('should allow inserting text sets', async () => {
+      const textSet = fixture.editor.library.getAllByRole('listitem', {
+        name: /^Insert Text Set$/,
+      })[0];
+
+      // The page should start off with no text elements
+      expect((await getTextElements()).length).toBe(0);
+
+      await fixture.events.mouse.clickOn(textSet, 25, 25);
 
-  // Disable reason: not implemented yet.
-  // eslint-disable-next-line jasmine/no-disabled-tests
-  xit('should allow inserting text sets', () => {});
+      // After text set has been added, there should some text elements
+      expect((await getTextElements()).length).toBeGreaterThan(0);
+    });
+
+    it('should allow user to drag and drop text set onto page', async () => {
+      const textSet = fixture.editor.library.getAllByRole('listitem', {
+        name: /^Insert Text Set$/,
+      })[0];
+
+      const page = fixture.editor.canvas.fullbleed.container;
+
+      // The page should start off with no text elements
+      expect((await getTextElements()).length).toBe(0);
+
+      await fixture.events.mouse.moveRel(textSet, 25, 25);
+      await fixture.events.mouse.down();
+
+      await fixture.events.mouse.moveRel(page, 50, 100);
+      await fixture.events.mouse.up();
+
+      // After text set has been added, there should some text elements
+      expect((await getTextElements()).length).toBeGreaterThan(0);
```

Comment:
Please add a snapshot here with `await fixture.snapshot('Text set added')`.

We could even do a snapshot of the text set as it's being dragged to verify the preview display. Maybe just before `mouse.up`?